sdk/core/core-client/test/serializer.spec.ts

```diff
@@ -74,6 +74,28 @@ describe("Serializer", function () {
       assert.deepEqual(serialized, expected);
     });
 
+    it.only("should not set default value for non-required properties", function () {
+      const expected = {
+        id: 1,
+        name: "testProduct",
+        details: {
+          max_product_capacity: "Large",
+        },
+      };
+
+      const serialized = Serializer.serialize(
```

Comment:
need a deserialize test rather than a serialize one

Reply:
Thanks for pointing it out :)
silly me :joy:

Reply:
It took me a little bit to notice as well 😄